src/Tickets/Emails/Email_Abstract.php

```diff
@@ -434,8 +434,8 @@ public function get_placeholders(): array {
 	 */
 	public function format_string( $string ): string {
 		$placeholders = $this->get_placeholders();
-		$find         = array_keys( $placeholders );
-		$replace      = array_values( $placeholders );
+		$find         = array_keys( $this->placeholders );
+		$replace      = array_values( $this->placeholders );
```

Comment:
The point of getting the placeholders from `get_placeholders()`, is that it is returning the filtered placeholders, so we can hook from other places. [Such as TEC](https://github.com/the-events-calendar/the-events-calendar/pull/4196/files#diff-7a9e4d535269e6b251e2a56a46c6e38ae4d0bcb385b87def7227e79e43664337R33). We should keep it like that and revert this change.

Reply:
Ah. I was getting an error. Looking back at it now, I see it was because of the `@return` type for the `get_placeholders` method. Sorry about that. I'll fix it here.